File: movieshows/src/main/java/com/niranjandroid/movieshows/app/AppModule.java
package com.niranjandroid.movieshows.app;

import android.app.Application;
import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by Niranjan P on 4/18/2017.
 */

@Module
public class AppModule {

    Application application;

    public AppModule(Application application) {
        this.application = application;
    }

    @Provides
    @Singleton
    public Application providesApplication() {
        return application;
    }

    @Provides
    @Singleton
    public Context provideApplicationContext() {
        return application.getApplicationContext();
    }
}

File: movieshows/src/main/java/com/niranjandroid/movieshows/app/AppModule.kt
package com.niranjandroid.movieshows.app

import android.app.Application
import android.content.Context
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Created by Niranjan P on 4/18/2017.
 */

@Module
class AppModule(internal var application: Application) {

    @Provides
    @Singleton
    fun providesApplication(): Application {
        return application
    }

    @Provides
    @Singleton
    fun provideApplicationContext(): Context {
        return application.applicationContext
    }
}
